nep-11.mediawiki

```diff
@@ -0,0 +1,231 @@
+
+If the method succeeds, it MUST fire the <code>Transfer</code> event, and MUST return <code>true</code>, even if the <code>amount</code> is <code>0</code>, or the token is sent to the owner.
+
+The function SHOULD check whether the <code>from</code> address equals the caller contract hash.
+If so, the transfer SHOULD be processed; If not, the function SHOULD use the SYSCALL <code>Neo.Runtime.CheckWitness</code> to verify the transfer.
+
+If the <code>to</code> address is a deployed contract, the function SHOULD check the <code>payable</code> flag of this contract to decide whether it should transfer the tokens to this contract.
+
+If the transfer is not processed, the function SHOULD return <code>false</code>.
+
+====ownerOf====
+
+<pre>
+public static enumerator ownerOf(byte[] tokenid)
+</pre>
+
+Returns an <code>enumerator</code> that contains all the co-owners that own the specified token.
+
+The parameter <code>tokenid</code> SHOULD be a valid NFT.
+If not, this method SHOULD <code>throw</code> an exception.
+
+====balanceOf====
+
+<pre>
+public static BigInteger balanceOf(byte[] owner)
+</pre>
+
+Returns the total amount of NFTs owned by the specified address.
+
+The parameter <code>owner</code> SHOULD be a 20-byte address.
+If not, this method SHOULD <code>throw</code> an exception.
+
+<pre>
+public static BigInteger balanceOf(byte[] owner, byte[] tokenid)
+</pre>
+
+If the second parameter <code>tokenid</code> is also provided, this method returns the balance of the specified token for the specified owner's account.
+
+The parameter <code>tokenid</code> SHOULD be a valid NFT.
+If not, this method SHOULD <code>throw</code> an exception.
+
+The parameter <code>owner</code> SHOULD be a 20-byte address.
+If not, this method SHOULD <code>throw</code> an exception.
+
+If the <code>owner</code> is an unused address, or it's not the owner of the specified token, this method SHOULD return <code>0</code>.
+
+====tokensOf====
+
+<pre>
+public static enumerator tokensOf(byte[] owner)
+</pre>
+
+Returns an <code>enumerator</code> that contains all the tokens owned by the specified address.
+
+The parameter <code>owner</code> SHOULD be a 20-byte address.
+If not, this method SHOULD <code>throw</code> an exception.
+
+The returned tokens should be in the following format:
+<pre>
+{
+  "id": <ID>,
+  "properties": properties(<ID>),
+  ...
+}
+</pre>
+
+It must have the fields <code>id</code> and <code>properties</code>.
+It can contain other dApp-specific fields.
+
+====properties====
+
+<pre>
+public static string properties(byte[] tokenid)
+</pre>
+
+Returns a serialized NVM object containing the properties for the given NFT.
+The NVM object must conform to the "NEO NFT Metadata JSON Schema".
+
+The parameter <code>tokenid</code> SHOULD be a valid NFT.
+If not, this method SHOULD <code>throw</code> an exception.
+
+====Metadata JSON Schema====
+
+<pre>
+{
+  "title": "Asset Metadata",
+  "type": "object",
+  "properties": {
```

Comment:
Since we have removed the `tokenURI()` from methods, we should add a new similar field into the "Metadata JSON Schema". This new field could be optional.

Reply:
Why?

Reply:
For example, each NFT can have a web page that can display some details or pictures.

Reply:
But do we need to specify a specific field? Isn't there the optional `image` field? Or `tokenURI` would kind of be a primary URI for that token?

Reply:
We've removed `tokenURI()`. So I'm asking to add it as a field of "Metadata JSON Schema", which will be returned by `properties()`.

Reply:
So one of the items in `properties` in the schema will be `tokenURI`?

Reply:
Yes.